docs: add new tutorial unit two (#3298)
refinedev/refine: documentation/docs/tutorial/1-getting-started/antd/3-generate-crud-pages.md

1- The goal of Inferencer is to reduce the amount of time spent on creating views for resources by generating the code automatically. This can save developers a significant amount of time and effort, especially on large projects with many resources.

2- Inferencer generates code that is easy to customize. Instead of starting from scratch, you can use the auto-generated code as a starting point and make changes to fit your specific needs. This allows you to quickly iterate and get your project up and running faster.

3- By using Inferencer, you can avoid common mistakes that can arise when building CRUD pages manually. This can help you avoid bugs and other issues that can arise during development.

Overall, using Inferencer can greatly speed up development time and reduce the amount of code that needs to be written manually. It helps ensure that your CRUD pages are consistent and adhere to best practices, freeing you up to focus on more complex tasks.

#### Learn Inferencer

-   Learn about [how Inferencer works](/docs/packages/documentation/inferencer).
-   Learn about [how to use Inferencer with Ant Design](/docs/api-reference/antd/components/inferencer/).

> 

Reviewer: Here we can add section like 
### How to activate(use) Inferencer
Here you show registering Inferencer components to the `<Refine>` component like we did in [how to use Inferencer with Ant Design](/docs/api-reference/antd/components/inferencer/)

Author: @omeraplak said the inferencer will come directly added. What do @omeraplak think about?

Reviewer 2: ahh my fault :( Necati is right ⚡️

## Preview the generated pages

Now that you have created your project, let's take a look at the CRUD pages that Inferencer has generated for us.

> 

Reviewer: It'll be nice if you can mention that the app consumes https://api.fake-rest.refine.dev and so by using inferencer generated code you can now list, edit, create the records on the api. Also mention we'll see how refine communicates with the API in the data provider section..

2- Inferencer generates code that is easy to customize. Instead of starting from scratch, you can use the auto-generated code as a starting point and make changes to fit your specific needs. This allows you to quickly iterate and get your project up and running faster.

3- By using Inferencer, you can avoid common mistakes that can arise when building CRUD pages manually. This can help you avoid bugs and other issues that can arise during development.

Overall, using Inferencer can greatly speed up development time and reduce the amount of code that needs to be written manually. It helps ensure that your CRUD pages are consistent and adhere to best practices, freeing you up to focus on more complex tasks.

#### Learn Inferencer

-   Learn about [how Inferencer works](/docs/packages/documentation/inferencer).
-   Learn about [how to use Inferencer with Ant Design](/docs/api-reference/antd/components/inferencer/).

## How to use Inferencer

> The `@pankod/refine-inferencer` package includes the `<AntdInferencer />` component, which can be imported from `@pankod/refine-inferencer/antd` and used to generate CRUD pages with Ant Design components.

Reviewer: ```suggestion
The `@pankod/refine-inferencer` package provides the `<AntdInferencer />` component, which can be imported from `@pankod/refine-inferencer/antd` and used to generate CRUD pages based on your API response with Ant Design components.
```

> The component can be used directly in the `resources` prop of the `<Refine />` component in `App.tsx` as shown below:

Reviewer: ```suggestion
The `<AntdInferencer />` component is used by passing to appropriate values in the `resources` prop of the `<Refine />` component in `App.tsx` as shown below:
```

> The component can be used directly in the `resources` prop of the `<Refine />` component in `App.tsx` as shown below:

Reviewer: ```suggestion
The `<AntdInferencer />` component is used by passing to appropriate values in the `resources` prop of the `<Refine />` component in `App.tsx` as shown below:
```

:::info

> In [Unit 4](/docs/tutorial/understanding-resources/index), resource concepts will be explained in detail. For now, you can think of a resource as a collection of data that you want to manage in your app.

Reviewer: ```suggestion
In [Unit 4](/docs/tutorial/understanding-resources/index), the resources concept will be explained in detail. For now, you can assume that the resource is a collection of data from your API used in the app.
```

:::

```tsx title="src/App.tsx"
import { Refine } from "@pankod/refine-core";
import {
    Layout,
    ReadyPage,
    notificationProvider,
    ErrorComponent,
} from "@pankod/refine-antd";
import routerProvider from "@pankod/refine-react-router-v6";
import dataProvider from "@pankod/refine-simple-rest";
//highlight-next-line
import { AntdInferencer } from "@pankod/refine-inferencer/antd";

import "@pankod/refine-antd/dist/reset.css";

const App: React.FC = () => {
    return (
        <Refine
            routerProvider={routerProvider}
            dataProvider={dataProvider("https://api.fake-rest.refine.dev")}
            Layout={Layout}
            ReadyPage={ReadyPage}
            notificationProvider={notificationProvider}
            catchAll={<ErrorComponent />}
            //highlight-start
            resources={[
                {
                    name: "posts",
                    list: AntdInferencer,
                    show: AntdInferencer,
                    create: AntdInferencer,
                    edit: AntdInferencer,
                },
            ]}
            //highlight-end
        />
    );
};

export default App;
```

> 

Reviewer: ```suggestion
At this point, refine will auto-generate the CRUD pages based on the API response on the fly
```

> ## Preview the generated pages

Reviewer: ```suggestion
## Preview the auto-generated pages
```

> Now that you have created your project, let's take a look at the CRUD pages that Inferencer has generated for us.

Reviewer: ```suggestion
So far we have added complete CRUD pages by utilizing Inferencer feature to the project. Let's take a look at the auto-generated CRUD pages that Inferencer handles for us.
```

> Before we start, let's understand the API that we will be using in this tutorial. We will be using the [JSONPlaceholder](https://jsonplaceholder.typicode.com/) API, which is a created by **refine** team to provide a fake REST API for testing. You can check out the **refine** fake REST API: https://api.fake-rest.refine.dev

Reviewer: ```suggestion
Before we start, let's understand the API that we will be using in this tutorial. We will be using the [JSONPlaceholder](https://jsonplaceholder.typicode.com/) API, which is created by **refine** team to provide a fake REST API for testing purposes. 
 
 You can check out the **refine** fake REST API: https://api.fake-rest.refine.dev
```

:::info

In [Unit 3](/docs/tutorial/understanding-dataprovider/index), it will be explained in detail how **refine** apps communicate with the API via the `dataProvider`.

:::

> 

Reviewer: By clicking to "Show Code" button in the live previews below, you can view the auto-generated codes by Inferencer.

```tsx live previewOnly previewHeight=600px url=http://localhost:3000/posts/show/123
setInitialRoutes(["/posts/show/123"]);

import { Refine } from "@pankod/refine-core";
import {
    Layout,
    ReadyPage,
    notificationProvider,
    ErrorComponent,
} from "@pankod/refine-antd";
import routerProvider from "@pankod/refine-react-router-v6";
import dataProvider from "@pankod/refine-simple-rest";
import { AntdInferencer } from "@pankod/refine-inferencer/antd";

import "@pankod/refine-antd/dist/reset.css";

const App: React.FC = () => {
    return (
        <Refine
            routerProvider={routerProvider}
            dataProvider={dataProvider("https://api.fake-rest.refine.dev")}
            Layout={Layout}
            ReadyPage={ReadyPage}
            notificationProvider={notificationProvider}
            catchAll={<ErrorComponent />}
            resources={[
                {
                    name: "posts",
                    list: AntdInferencer,
                    show: AntdInferencer,
                    create: AntdInferencer,
                    edit: AntdInferencer,
                },
            ]}
        />
    );
};

render(<App />);

> ```

Reviewer: At the end of the Inferencer section, you can say something like, we'll implement the auto generated page codes to our app in the next sections..
